src/Compilers/CSharp/Portable/Symbols/Compilation_WellKnownMembers.cs

```diff
@@ -471,19 +471,25 @@ internal SynthesizedAttributeData SynthesizeDynamicAttribute(TypeSymbol type, in
             Debug.Assert((object)type != null);
             Debug.Assert(type.ContainsDynamic());
 
+            MethodSymbol ctorSymbol;
+            ImmutableArray<TypedConstant> arguments;
+
             if (type.IsDynamic() && refKindOpt == RefKind.None && customModifiersCount == 0)
             {
-                return TrySynthesizeAttribute(WellKnownMember.System_Runtime_CompilerServices_DynamicAttribute__ctor);
+                ctorSymbol = (MethodSymbol)GetWellKnownTypeMember(WellKnownMember.System_Runtime_CompilerServices_DynamicAttribute__ctor);
+                arguments = ImmutableArray<TypedConstant>.Empty;
             }
             else
             {
-                NamedTypeSymbol booleanType = GetSpecialType(SpecialType.System_Boolean);
-                Debug.Assert((object)booleanType != null);
+                ctorSymbol = (MethodSymbol)GetWellKnownTypeMember(WellKnownMember.System_Runtime_CompilerServices_DynamicAttribute__ctorTransformFlags);
+                var argumentType = (ArrayTypeSymbol)ctorSymbol.Parameters[0].Type;
+                var booleanType = argumentType.ElementType;
+                Debug.Assert(booleanType.SpecialType == SpecialType.System_Boolean);
                 var transformFlags = DynamicTransformsEncoder.Encode(type, booleanType, customModifiersCount, refKindOpt);
-                var boolArray = ArrayTypeSymbol.CreateSZArray(booleanType.ContainingAssembly, booleanType, customModifiers: ImmutableArray<CustomModifier>.Empty);
-                var arguments = ImmutableArray.Create<TypedConstant>(new TypedConstant(boolArray, transformFlags));
-                return TrySynthesizeAttribute(WellKnownMember.System_Runtime_CompilerServices_DynamicAttribute__ctorTransformFlags, arguments);
+                arguments = ImmutableArray.Create<TypedConstant>(new TypedConstant(argumentType, transformFlags));
             }
+
+            return new SynthesizedAttributeData(ctorSymbol, arguments, ImmutableArray<KeyValuePair<string, TypedConstant>>.Empty);
```

Comment:
The motivation for this change was to avoid calling ```Compilation.GetSpecialType(SpecialType.System_Boolean)``` which I believe was assumed could fail. However, by the time we get here, the caller has already verified that DynamicAttribute .ctor method is valid so System.Boolean should be defined as well. In short ```GetSpecialType(SpecialType.System_Boolean)``` should not fail here. 

An alternative fix is to leave this method unchanged but simply remove the invalid check from `AddSynthesizedAttributes` below since it will not fail after checking for ```DynamicAttribute..ctor```.